src/JuliusSweetland.OptiKey/UI/Controls/KeyboardHost.cs

```diff
@@ -408,7 +408,14 @@ private void TraverseAllKeysAndBuildPointToKeyValueMap()
 
                     if (rect.Size.Width != 0 && rect.Size.Height != 0)
                     {
-                        pointToKeyValueMap.Add(rect, key.Value);
+                        if (pointToKeyValueMap.ContainsKey(rect))
+                        {
+                            Log.Error("Duplicate key, cannot add to map");
```

Comment:
Can you amend to include the key value in the error message so it is easier to debug?